packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -304,6 +306,21 @@ See ${createDocumentationLink({
       })}`
     );
 
+    safelyRunOnBrowser(() => {
+      if (options.future?.preserveSharedStateOnUnmount === undefined) {
```

Comment:
why not `warn` or `warning`? they already take a boolean that can be filled with `options.future?.preserveSharedStateOnUnmount !== undefined`

Reply:
A warning might signify to users that not adopting the change will break their implementation, so I preferred a lighter informational message.

Reply:
I'm mainly thinking of the __DEV__ wrapping for warn. Not sure if info or warn makes much difference though, it's a valid point

Reply:
Oh right, I didn't think about this and agree it should be scoped to dev mode. I'll add this.